### PR TITLE
refactor(app): refactor app domain

### PR DESCRIPTION
**Pull request by Oxiang**
## Problem

Incomplete migration of JSX files to TSX files

## Solution

- convert jsx files to tsx files
- convert js file to ts file
- change redux higher order connect to redux hooks
- assign types to function components, styles and methods
- replace proptypes with typescript

### CODE REVIEW

File: src/client/app/components/BaseLayout/BaseLayoutHeader.tsx [hunk 1]
```diff
@@ -233,7 +240,7 @@ const BaseLayoutHeader = ({
             headers.map(
               (header) =>
                 (header.public ? !isLoggedIn : isLoggedIn) &&
-                !header.hidden && (
```
Comment (yong-jie) [inline src/client/app/components/BaseLayout/BaseLayoutHeader.tsx:236]: can i check what was the context behind this removal?

File: src/client/app/components/pages/RootPage/ScrollToTop.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+const ScrollToTop = ({ children }: ScrollToTopProps) => {
+  const location = useLocation().pathname
+  const prevProps = usePrevious(location)
+  useEffect(() => {
+    if (!!prevProps && location !== prevProps) {
+      window.scrollTo(0, 0)
+    }
+  })
+  return <>{children}</>
```
Comment (yong-jie) [inline src/client/app/components/pages/RootPage/ScrollToTop.tsx:24]: wondering if `return children` would work

File: src/client/app/components/pages/RootPage/index.tsx [hunk 3]
```diff
@@ -29,7 +28,11 @@ import {
 } from '../../../util/types'
 import theme from '../../../theme'
 
-const Root = ({ store }) => (
+type RootProps = {
+  store: any
```
Comment (yong-jie) [inline src/client/app/components/pages/RootPage/index.tsx:32]: is there a stricter subset of `any` that would work for root props? maybe the redux libraries export a store type that can be used here

**Review by yong-jie — CHANGES_REQUESTED** [review]

File: src/client/app/components/pages/RootPage/ScrollToTop.tsx (hunk 2 above)
Comment (yong-jie) [reply src/client/app/components/pages/RootPage/ScrollToTop.tsx:24]: okay nevermind scratch that, didnt work.

File: src/client/app/components/BaseLayout/BaseLayoutHeader.tsx (hunk 1 above)
Comment (Oxiang) [reply src/client/app/components/BaseLayout/BaseLayoutHeader.tsx:236]: Headers' objects does not have 'hidden' field, so this was not used

**Review by yong-jie — APPROVED** [review]

**Pull request merged**